src/Tribe/Views/V2/Utils/View.php

```diff
@@ -54,4 +54,35 @@ public static function get_past_event_display_key() {
 		return $event_display_key;
 	}
 
-}
\ No newline at end of file
+	/**
+	 * Cleans the View data that will be printed by the `components/data.php` template to avoid its mangling.
+	 *
+	 * By default, the View data is a copy of the View template variables, to avoid the mangling of the JSON data
+	 * some entries of the data might require to be removed, some might require to be formatted or escaped.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,string|array> $view_data The initial View data.
+	 *
+	 * @return array<string,string|array> The filtered View data, some entries removed from it to avoid the data script
+	 *                                    being mangled by escaping and texturizing functions running on it.
+	 */
+	public static function clean_data( $view_data ) {
+		if ( ! is_array( $view_data ) ) {
+			return $view_data;
+		}
+
+		/*
+		 * Remove the JSON-LD data, it's already printed by the `components/json-ld-data.php` template. Printing a
+		 * `<script>`, the JSON-LD data, inside a `<script>`, the data, will cause issues.
+		 */
+		$view_data = array_diff_key( $view_data, array_flip( [ 'json_ld_data' ] ) );
+
+		// Remove objects that should not be printed on the page, keep data objects.
+		$view_data = array_filter( $view_data, static function ( $value ) {
+			return ! is_object( $value ) || $value instanceof \stdClass;
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 9b3c6cc2e0752d173f1e3afff828cc10</sub>